pkg/operation/botanist/component/etcd/dependency_watchdog.go

```diff
@@ -15,32 +15,29 @@
 package etcd
 
 import (
-	restarterapi "github.com/gardener/dependency-watchdog/pkg/restarter/api"
+	weederapi "github.com/gardener/dependency-watchdog/api/weeder"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
 )
 
-// DependencyWatchdogEndpointConfiguration returns the configuration for the dependency watchdog ensuring that its dependant
+// DependencyWatchdogWeederConfiguration returns the configuration for the dependency watchdog ensuring that its dependant
 // pods are restarted as soon as it recovers from a crash loop.
-func DependencyWatchdogEndpointConfiguration(role string) (map[string]restarterapi.Service, error) {
-	return map[string]restarterapi.Service{
+func DependencyWatchdogWeederConfiguration(role string) (map[string]weederapi.DependantSelectors, error) {
```

Comment:
This looks like a constructor. Isn't it?
```suggestion
func NewDependencyWatchdogWeederConfiguration(role string) (map[string]weederapi.DependantSelectors, error) {
```